device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubReceiveTask.java

```diff
@@ -32,11 +40,24 @@ public void run()
 
         try
         {
-            // Codes_SRS_IOTHUBRECEIVETASK_11_002: [The function shall poll an IoT Hub for messages, invoke the message callback if one exists, and return one of COMPLETE, ABANDON, or REJECT to the IoT Hub.]
+            // HTTP is the only protocol where the SDK must actively poll for received messages. Because of that, never
+            // wait on the IoTHubTransport layer to notify this thread that a received message is ready to be handled.
+            if (this.transport.getProtocol() != IotHubClientProtocol.HTTPS)
+            {
+                synchronized (this.receiveThreadLock)
+                {
+                    if (!this.transport.hasReceivedMessagesToHandle() && !this.transport.isClosed())
+                    {
+                        // AMQP and MQTT layers will notify the IoTHubTransport layer once a message arrives, and at
+                        // that time, this thread will be notified to handle them.
+                        this.receiveThreadLock.wait();
```

Comment:
See [here](https://www.baeldung.com/java-wait-notify#the-wait-method) for a more in-depth explanation on wait/notify, but the short of it is that this thread goes dormant until notified. When dormant, this thread has minimal performance impact on the machine running the SDK